spring-kafka/src/main/java/org/springframework/kafka/core/KafkaTemplate.java

```diff
@@ -465,6 +465,15 @@ public void setObservationRegistry(ObservationRegistry observationRegistry) {
 		this.observationRegistry = observationRegistry;
 	}
 
+	/**
+	 * Return the {@link ObservationRegistry} used by the template.
+	 * @return the observation registry
+	 * @since 3.3.5
+	 */
+	public ObservationRegistry getObservationRegistry() {
```

Comment:
Is it really has to be `public`?
Isn't just `protected` enough for our requirements in the `ReplyingKafkaTemplate`?

Reply:
Good point, protected visibility is indeed enough.